### PR TITLE
refactor!: remove Testabiliy and TestabilityRegistry from minimal application and make it injectable

### PR DESCRIPTION
**Pull request by ykmnkmi**
`Testability` is the only API that relies on the `js` package (not `js_util`) and that increases the build size. Now, in order to use the `Testability` API, you need to inject it manually using `GenerateInjector` with `testabilityProvider`:
```dart
@GenerateInjector([testabilityProvider])
final InjectorFactory appInjector = ng.appInjector$Injector;

void main() {
  runApp(ng.MyAppComponentNgFactory, createInjector: appInjector);
}
```
For example, the build size of the `hello_world` (without `js` package) example was 163,315 and is now 101,983.

Also some ToDo fixes.

### CODE REVIEW

**Comment by ykmnkmi** [comment]
Or remove `Testability` API?

**Comment by GZGavinZhao** [comment]
Nice code size reduction!

I agree with this mostly, but maybe not remove it for now. I think it is still useful to be able to know the state (particularly the `isStable` stuff) from the JS side for testing & potentially SSR. Not automatically injecting it is enough.

File: ngdart/lib/src/testability/js_api.dart [hunk 1]
```diff
@@ -11,8 +12,8 @@ import 'package:meta/meta.dart';
 @anonymous
 abstract class JsTestability {
   external factory JsTestability({
-    @required bool Function() isStable,
-    @required void Function(void Function(bool didAsyncWork)) whenStable,
```
Comment (GZGavinZhao) [inline ngdart/lib/src/testability/js_api.dart:15]: Is there a reason why you removed the `didAsyncWork` parameter?

Comment (ykmnkmi) [reply ngdart/lib/src/testability/js_api.dart:15]: Looking at `// TODO(b/168535057): Remove \`didWork\`.` comment, I delete them all, although it was necessary to delete only [there](https://github.com/angulardart-community/angular/blob/master/ngdart/lib/src/testability/js_impl.dart#L108). I can restore them.

Comment (GZGavinZhao) [reply ngdart/lib/src/testability/js_api.dart:15]: It's fine :)

**Review by GZGavinZhao — APPROVED** [review]

**Pull request merged**